driver/normalizer/normalizer.go

```diff
@@ -28,31 +28,53 @@ var Normalize = Transformers([][]Transformer{
 
 func funcDefMap(typ string, async bool) Mapping {
 	return MapSemantic(typ, uast.FunctionGroup{}, MapObj(
-		Obj{
-			"body": Var("body"),
-			"name": Var("name"),
+		Fields{
+			{Name: "body", Op: Var("body")},
+			{Name: "name", Op: Var("name")},
 			// Arguments should be converted by the uast.Arguments normalization
-			"args": Obj{
+			{Name: "args", Op: Obj{
 				"args":       Var("arguments"),
 				uast.KeyPos:  Var("_pos"),
 				uast.KeyType: Var("_type"),
+			}},
+			// Will be filled only if there is a Python3 type annotation
+			{Name: "returns", Optional: "ret_opt", Op: Cases("ret_case",
+				Is(nil),
+				Obj{
+					uast.KeyType:  String("BoxedName"),
+					"boxed_value": Var("ret_type"),
+					// No problem dropping this one, it's used by an internal interpreter optimization/cache
+					// without semantic meaning
+					"ctx": Any(),
+				}),
 			},
+			{Name: "decorator_list", Op: Var("func_decorators")},
 		},
 		Obj{
 			"Nodes": Arr(
 				Obj{
 					// FIXME: generator=true if it uses yield anywhere in the body
 					"async": Bool(async),
 				},
+				Obj{
+					"decorators": Var("func_decorators"),
+				},
 				UASTType(uast.Alias{}, Obj{
 					// FIXME: can't call identifierWithPos because it would take the position of the
 					// function node that is not exactly the same as the position of the function name
 					"Name": UASTType(uast.Identifier{}, Obj{
 						"Name": Var("name"),
 					}),
 					"Node": UASTType(uast.Function{}, Obj{
-						"Type": UASTType(uast.FunctionType{}, Obj{
-							"Arguments": Var("arguments"),
+						"Type": UASTType(uast.FunctionType{}, Fields{
+							{Name: "Arguments", Op: Var("arguments")},
+							{Name: "Returns", Optional: "ret_opt", Op: Cases("ret_case",
+								Is(nil),
```

Comment:
Python functions always have an implicit return, so should be a single empty `Argument` with `Init` = `None` when there are no type annotations. There is a similar piece of code in [JS](https://github.com/bblfsh/javascript-driver/blob/master/driver/normalizer/normalizer.go#L274)).

Reply:
I implemented this and it worked beautifully... but then I noticed that this annotations only has a value IF the function has a type annotation BUT if it doesn't have it, it doesn't mean that the function returns None! Just that it's not annotated. So I think is better to left it with the current state (nil if there is no annotation, the annotation if there is one).

Reply:
I haven't meant that the function returns `None` as type, only that if there is no return, all Python2 and Python3 functions returns `None` implicitly. So it's `Returns.Argument.Init: None`, same way as we do it in JS with `undefined`.

Reply:
Yes, but to know if the function returns None implicitly we should parse the function and find the return statement in this case.